Userland/Libraries/LibJS/Runtime/SymbolConstructor.cpp

```diff
@@ -57,10 +57,10 @@ ThrowCompletionOr<NonnullGCPtr<Object>> SymbolConstructor::construct(FunctionObj
 JS_DEFINE_NATIVE_FUNCTION(SymbolConstructor::for_)
 {
     // 1. Let stringKey be ? ToString(key).
-    auto string_key = TRY(vm.argument(0).to_deprecated_string(vm));
+    auto string_key = TRY(vm.argument(0).to_string(vm));
 
     // 2. For each element e of the GlobalSymbolRegistry List, do
-    auto result = vm.global_symbol_registry().get(string_key);
+    auto result = vm.global_symbol_registry().get(string_key.to_deprecated_string());
```

Comment:
hmm is `to_deprecated_string()` necessary here? I believe `String` and `DeprecatedString` are hash-compatible

Reply:
`HashMap<K, V>::get()` wants a `K const&` and this is a `<DeprecatedString, NonnullGCPtr<Symbol>>`, so yes it's needed.

Reply:
This *should* work because there are several `get` overloads, a few of which have a templated key type constrained by a hash-compatible trait. But looks like it fails to compile later on just because there is no `operator==` for String and DeprecatedString.

We could A) add that definition, B) use `bytes_as_string_view()` here to avoid the allocation, but this is all temporary anyways so meh.

Reply:
I just updated that hashmap to use new strings, which it obviously can do very easily now. Problem solved :galaxybrain: